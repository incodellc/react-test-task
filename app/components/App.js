import '../styles/application.scss';
import React, { PureComponent } from 'react';
import Ui from './Ui';
import { connect } from '../services/tickerService';

// The below line is here as an example of getting prices
// connect('AAPL');

class App extends PureComponent {
    state = {
        stockSymbol: 'AAPL',
        time: 500,
        data: null,
        loading: true
    }
    componentDidMount() {
        connect(this.state.stockSymbol, this.onLoadData);
    }
    onLoadData = (data) => {
        const newData = JSON.parse(data);
        this.setState({
            data: newData,
            loading: false,
        });
    }

    render() {
        const { data, loading } = this.state;
        if (data === null && loading === true) {
            return (<p className="green">Loading...</p>);
        }
        if (( typeof data !== 'object' || data === null || Object.keys(data).length === 0 ) && loading === false) {
            return (<p className="red">Error!!!!!!!!</p>);
        }
        return (<div className="stock-ticker">
            <h1>Stock Blotter</h1>
            <Ui data={this.state.data} time={this.state.time}/>
            </div>);
    }
}


export default App;
