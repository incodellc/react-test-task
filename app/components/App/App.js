import '../../styles/application.scss';
import React from 'react';
import TickersContainer from '../../containers/TickersContainer/TickersContainer';
import Header from '../Header/Header';

const App = () => {
    return (
        <div className="stock-ticker">
            <div className="container">
                <Header />
                <TickersContainer />
            </div>
        </div>
    );
};

export default App;
