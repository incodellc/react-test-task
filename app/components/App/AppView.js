import React from 'react';
import T from 'prop-types';
import '../../styles/application.scss';
import PriceList from './components/List/PriceListView';

// The below line is here as an example of getting prices
// connect('AAPL'); priceOperations,
// const ticker = 'APPL';

function App({ isLoading, isStartFetch, handleTickerChange,
    changePrice, handleTickerChangeTime, changeTime,
    seconds, value, List, chartData }) {
    if(isLoading) {
        return (
            <div>Loading...</div>
        );
    }
    return (
        <div className="stock-ticker font-initial">
            <header className="stock-header">
                <h1>Stock Blotter</h1>
            </header>
            {(isStartFetch) ? <main className="stock-ticker_symbol">
                <div className="stock-choice_font">
                    Select Time interval:
                </div>
                <div>
                    <select className="satock-select font-initial"
                    onChange={
                        (evt) => handleTickerChangeTime(evt.target.value)}
                        value={seconds}
                    >
                        <option value="1000">1 second</option>
                        <option value="2000">2 seconds</option>
                        <option value="3000">3 seconds</option>
                        <option value="4000">4 seconds</option>
                        <option value="5000">5 seconds</option>
                        <option value="10000">10 seconds</option>
                    </select>
                    <button className="stock-button font-initial"
                    type="button"
                    onClick={changeTime}>
                        Change Interval
                    </button>
                </div>
            </main> : <main className="stock-ticker_symbol">
                <div className="stock-choice_font">
                    Please select stock symbol and put "get started"
                </div>
                <div>
                    <select className="satock-select font-initial" onChange={(evt) => handleTickerChange(evt.target.value)} value={value}>
                        <option value="APPL">Inc.(APPL)</option>
                        <option value="BEST">Inc.(BEST)</option>
                        <option value="DOW">Inc.(DOW)</option>
                        <option value="HAS">Inc.(HAS)</option>
                        <option value="OVV">Inc.(OVV)</option>
                        <option value="AMZN">Inc.(AMZN)</option>
                        <option value="NIO">Inc.(NIO)</option>
                        <option value="ACB">Inc.(ACB)</option>
                    </select>
                    {/* Or using handle input for enter stock symbol */}
                    {/* <input type="text"
                onChange={(evt) => handleTickerChange(evt.target.value)}
                value={value}/> */}
                    <button className="stock-button font-initial"
                    type="button"
                    onClick={changePrice}>
                        Get Started
                    </button>
                </div>
            </main>}
            <PriceList list={List} chartData={chartData} />
        </div>
    );
}
export default App;
App.propTypes = {
    isLoading: T.bool.isRequired,
    isStartFetch: T.bool.isRequired,
    List: T.object.isRequired,
    handleTickerChange: T.func.isRequired,
    changePrice: T.func.isRequired,
    handleTickerChangeTime: T.func.isRequired,
    changeTime: T.func.isRequired,
    seconds: T.string.isRequired,
    value: T.string.isRequired,
    chartData: T.array,
};
