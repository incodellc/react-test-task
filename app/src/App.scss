* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
.App {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
  .row {
    .name {
      padding-right: 15px;
      text-transform: capitalize;
    }
    .value {
      font-weight: bold;
      padding-right: 5px;
    }
    .arrow {
      font-weight: bold;
      &.arrow-up {
        color: green;
      }
      &.arrow-down {
        color: red;
      }
    }
  }
}
