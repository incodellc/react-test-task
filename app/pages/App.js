import React from 'react';
import { StockPage } from './StockPage';
import { MainContainer } from './App.styles';

export const App = () => {
    return (
        <MainContainer>
            <StockPage/>
        </MainContainer>
    );
};

export default App;
