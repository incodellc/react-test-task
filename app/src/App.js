import React from "react";
import "./App.scss";
import { Provider } from "react-redux";
import { store } from "./store/store";
import { SingleTicker } from "./components/singleTicket";

export const App = () => (
  <Provider store={store}>
    <div className="App">
      <SingleTicker />
    </div>
  </Provider>
);
