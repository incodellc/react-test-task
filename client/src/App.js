import React, { useEffect, useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { connectToStreamServer } from './store/actions/action-creators';
import { isConnectedSelector } from './store/selectors';
import { Sidebar, Segment } from 'semantic-ui-react';
import { Layout, TickerList } from './components';

const App = () => {
    const dispatch = useDispatch();
    const isConnected = useSelector(isConnectedSelector); 
    const connectToStreamServerHandler = useCallback(() => dispatch(connectToStreamServer()), [dispatch]);

    useEffect(() => connectToStreamServerHandler(), [connectToStreamServerHandler]);

    return (
        <Layout>
            <Sidebar.Pushable>
                <Sidebar
                    animation='push'
                    visible={true}
                >
                    <TickerList />
                </Sidebar>
                <Sidebar.Pusher>
                    <Segment 
                        basic
                        style={{marginRight: '18.5714rem'}}
                    >

                    </Segment>
                </Sidebar.Pusher>
            </Sidebar.Pushable>
        </Layout>
    );
};

export default App;