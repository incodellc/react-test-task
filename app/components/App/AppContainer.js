import { connect } from 'react-redux';
import { compose, withState, withHandlers, lifecycle } from 'recompose';
import { priceOperations } from '../../reducers/price';
import socketApi from '../../services';
import AppView from './AppView';

const mapDispatchToProps = {
    fetchPrice: priceOperations.fetchPrice,
    handlePriceRealTime: priceOperations.handlePriceRealTime,
};
const mapStateToProps = (state) => ({
    List: state.price.latest.price,
    isLoading: state.price.latest.isloading,
    isStartFetch: state.price.latest.isStartFetch,
    chartData: state.price.latest.chartData,
});
const enhancer = compose(
    connect(
        mapStateToProps,
        mapDispatchToProps,
    ),
    withState('value', 'handleTickerChange', 'APPL'),
    withState('seconds', 'handleTickerChangeTime', '2000'),
    withHandlers({
        changePrice: (props) => () => {
            props.fetchPrice(props.value);
            props.handlePriceRealTime(props.value);
        },
        changeTime: (props) => () => {
            socketApi.setTime(props.seconds);
        },
    }),
    lifecycle({
        componentDidMount() {
            this.props.fetchPrice(this.props.value);
        },
    }),
);
export default enhancer(AppView);
