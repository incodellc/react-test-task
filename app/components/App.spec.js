import React from 'react';
import {mount, shallow} from 'enzyme';
import {Provider} from 'react-redux';
import App from './App';
import { mockStore } from '../../config/setupTest';

const store = mockStore({
    stockTicker: {
        currentData: {
            ticker: 'MNOV',
            exchange: 'NASDAQ',
            price: '-81.07',
            change: '-23.43',
            change_percent: '0.65',
            last_trade_time: '2020-09-27T07:49:32.000Z',
            dividend: '0.09',
            yield: '1.22'
        }
    }
});

it('should contain .stock-wrapper', () => {
    const wrapper = mount(
        <Provider store={store}>
            <App />
        </Provider>
    );
    expect(wrapper.find('.stock-wrapper').length).toBe(1);
});

it('should render App component with props', () => {
    const props = {
        currentData: {
            ticker: 'MNOV',
            exchange: 'NASDAQ',
            price: '-81.07',
            change: '-23.43',
            change_percent: '0.65',
            last_trade_time: '2020-09-27T07:49:32.000Z',
            dividend: '0.09',
            yield: '1.22'
        },
        saveTickerData: () => {},
    };

    const component = shallow(<Provider store={store}>
        <App {...props}/>
    </Provider>);
    expect(component).toMatchSnapshot();
});

it('should render App component without props', () => {
    const component = shallow(<Provider store={store}>
        <App />
    </Provider>);
    expect(component).toMatchSnapshot();
});
