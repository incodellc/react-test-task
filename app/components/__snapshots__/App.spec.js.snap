// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component with props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(App)
    currentData={
      Object {
        "change": "-23.43",
        "change_percent": "0.65",
        "dividend": "0.09",
        "exchange": "NASDAQ",
        "last_trade_time": "2020-09-27T07:49:32.000Z",
        "price": "-81.07",
        "ticker": "MNOV",
        "yield": "1.22",
      }
    }
    saveTickerData={[Function]}
  />
</ContextProvider>
`;

exports[`should render App component without props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(App) />
</ContextProvider>
`;
